### PR TITLE
logging

### PR DESCRIPTION
**Pull request by mahesh-hegde**
Last one for today.

### CODE REVIEW

File: jnigen/lib/src/logging/logging.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+enum LogLevel {
```
Comment (dcharkes) [inline jnigen/lib/src/logging/logging.dart:7]: We already have these in package logging. Why do we need a duplicate abstraction?

**Review by dcharkes — COMMENTED** [review]
Do we need a single integration test to see that the logging works as intended? E.g. run our own package with `Process` and capture stdout or stderr to see that it is working as expected.

File: jnigen/lib/src/logging/logging.dart (hunk 1 above)
Comment (mahesh-hegde) [reply jnigen/lib/src/logging/logging.dart:7]: I didn't know there was a standard logging package, and defined logging levels + logger. I wrote this class more similar to android's logging levels.

I should probably update this. I will update this PR tomorrow.

Re: unit test, I think it's more effort than worth, we have to update expected output every time we add a logging statement in code.

Comment (dcharkes) [reply jnigen/lib/src/logging/logging.dart:7]: Yeah, maybe leave the unit test out, unless you for example use the logger to error on the --config path not being found. That we wouldn't change that so often.

File: jnigen/lib/src/tools/android_sdk_tools.dart [hunk 2]
```diff
@@ -68,28 +73,34 @@ task listDependencies(type: Copy) {
...
+          (androidProject == '.') ? '' : ' in $androidProject';
       throw Exception('\n\ngradle exited with exit code ${procRes.exitCode}\n'
           'This can be related to a known issue with gradle. Please run '
-          '`flutter build apk` in $androidProject and try again\n');
+          '`flutter build apk`$inAndroidProject and try again\n');
     }
-    final gradleClassPaths = (procRes.stdout as String).split('\n');
-    if (gradleClassPaths.last.isEmpty) {
-      gradleClassPaths.removeLast();
+    final classpath = (procRes.stdout as String).split('\n');
```
[the remark this answers is not in the document]
Comment (mahesh-hegde) [reply jnigen/lib/src/tools/android_sdk_tools.dart:99]: conventionally classpath is set of paths. But I agree in some places in jnigen this terminology is inconsistent.

[the remark this answers is not in the document]
Comment (mahesh-hegde) [reply jnigen/lib/src/tools/android_sdk_tools.dart:99]: nevermind, in this context classpaths makes more sense.

**Review by dcharkes — APPROVED** [review]

**Pull request merged**